Model/Paypal_Payment_Method.php

```diff
@@ -93,4 +91,44 @@ public function getConfigFields()
 
         return array_merge($parentConfigFields, $additionalFields);
     }
+
+    /**
+     * @inheritdoc
+     *
+     * @return Transaction
+     */
+    public function getCancelTransaction(): Transaction
```

Comment:
I don't see the point of having this in the abstract payment method where you just call the $this->getNewTransactionInstance(); and in each payment method class only this getNewTransactionInstance where you return in this case new PayPalTransaction(). In the future you will only have 2 payment methods where you need to set more then parent transaction id. And there you can just override the parent method.

Reply:
ad `TransactionHandler.php`:
Removed the states from there entirely now and use the ones from the `Transaction` class.

ad `Paypal_Payment_Method.php`:
Alright, that makes sense, now only the `getTransaction()` method in the `PaymentMethod` class is abstract, and all the other transaction type getters directly return that method. So now we only need to override the parent method when needed like you suggested :)